zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/distribution/CommandDistributionIdempotencyTest.java

```diff
@@ -510,14 +511,17 @@ public static Collection<Object[]> scenarios() {
           },
           {
             "User.DELETE is idempotent",
-            new Scenario(
+            new Scenario<UserRecord>(
                 ValueType.USER,
                 UserIntent.DELETE,
                 () -> {
-                  final var user = createUser();
-                  ENGINE.user().deleteUser(user.getValue().getUsername()).delete();
+                  final var user = createUser("myUser");
```

Comment:
I needed to make the userName explicit here, thus I can match on that during acknowledgement handling. Alternative would be to create a `scenario.prepare` callback that stores the result record with the dynamic userName 💭